manifests/init.pp

```diff
@@ -17,6 +19,20 @@
     purge   => $config_dir_purge,
   }
 
+  if str2bool($cron_purge) {
+    file {'/etc/cron.hourly/00awstats':
+      ensure => 'absent'
```

Comment:
Puppet will do that correct thing, the resource doesn't need to be declared twice.